components/apps_modal/dialog_element/dialog_element.tsx

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {FormattedMessage} from 'react-intl';
+
+import {AppField} from 'mattermost-redux/types/apps';
+import {Channel} from 'mattermost-redux/types/channels';
+import {UserProfile} from 'mattermost-redux/types/users';
+
+import MenuActionProvider from 'components/suggestion/menu_action_provider';
+import GenericUserProvider from 'components/suggestion/generic_user_provider.jsx';
+import GenericChannelProvider from 'components/suggestion/generic_channel_provider.jsx';
+
+import TextSetting, {InputTypes} from 'components/widgets/settings/text_setting';
+import AutocompleteSelector from 'components/autocomplete_selector';
+import ModalSuggestionList from 'components/suggestion/modal_suggestion_list.jsx';
+import BoolSetting from 'components/widgets/settings/bool_setting';
+import RadioSetting from 'components/widgets/settings/radio_setting';
+import Provider from 'components/suggestion/provider';
+
+const TEXT_DEFAULT_MAX_LENGTH = 150;
+const TEXTAREA_DEFAULT_MAX_LENGTH = 3000;
+
+type StaticOption = {text: string; value: string};
+
+export type Props = {
+    field: AppField;
+    displayName: string;
+    name: string;
+    type: string;
+    subtype?: string;
+    placeholder?: string;
+    helpText?: string;
+    errorText?: React.ReactNode;
+    minLength?: number;
+    maxLength?: number;
+    dataSource?: string;
+    optional?: boolean;
+    options: {text: string; value: string}[] | null;
+    value: string | boolean | number | null;
+    onChange: (name: string, value: any) => void;
+    autoFocus?: boolean;
+    actions: {
+        autocompleteChannels: (term: string, success: (channels: Channel[]) => void, error: () => void) => (dispatch: any, getState: any) => Promise<void>;
+        autocompleteUsers: (search: string) => Promise<UserProfile[]>;
+    };
+}
+
+type State = {
+    value?: string;
+};
+
+export default class DialogElement extends React.PureComponent<Props, State> {
```

Comment:
Is there any chance to reuse the dialog element used for interactive dialogs? Specially the rendering seems pretty similar to the one used there.

Reply:
Update: Daniel and I discussed offline that we will keep the existing `InteractiveDialog` and `DialogElement` components the same to avoid regressions

Reply:
For terminology sake, can we please refer to Apps Modal Forms as "Modals"?